specification/storagecache/resource-manager/Microsoft.StorageCache/stable/2020-10-01/storagecache.json

```diff
@@ -1359,9 +1361,249 @@
+          "pattern": "^[-0-9a-zA-Z_]{1,15}$",
+          "type": "string"
+        },
+        "domainJoined": {
+          "description": "This field indicates if the HPC Cache is joined to the Active Directory Domain.",
+          "type": "string",
+          "readOnly": true,
+          "enum": [
+            "Yes",
+            "No",
+            "Error"
+          ],
+          "x-ms-enum": {
+            "name": "DomainJoinedType",
+            "modelAsString": true
+          }
+        },
+        "credentials": {
+          "description": "Active Directory admin or user credentials used to join the HPC Cache to a Domain.",
+          "properties": {
+            "username": {
+              "description": "User name of the Active Directory domain administrator. This value is stored encrypted and not returned on response.",
+              "type": "string"
+            },
+            "password": {
+              "description": "Plain text password of the Active Directory domain administrator. This value is stored encrypted and not returned on response.",
+              "x-ms-secret": true,
+              "type": "string"
+            }
+          },
+          "required": [
+            "username",
+            "password"
+          ]
+        }
+      },
+      "type": "object",
+      "required": [
+        "primaryDnsIpAddress",
+        "secondaryDnsIpAddress",
+        "domainName",
+        "smbServerName"
+      ]
+    },
+    "CacheUsernameDownloadSettings": {
+      "description": "Settings for username and group download for Extended groups.",
+      "type": "object",
+      "properties": {
+        "extendedGroups": {
+          "type": "boolean",
+          "description": "This indicates if Extended Groups is enabled."
+        },
+        "usernameSource": {
+          "description": "This setting determines how the system gets username and group names for clients.",
+          "enum": [
+            "AD",
+            "LDAP",
+            "File",
+            "None"
+          ],
+          "x-ms-enum": {
+            "name": "UsernameSource",
+            "modelAsString": true
+          },
+          "type": "string",
+          "default": "None"
+        },
+        "groupFileURI": {
+          "type": "string",
+          "description": "The URI of the file containing the group (etc/group) information. This field must be populated when 'usernameSource' is set to 'File'."
+        },
+        "userFileURI": {
+          "type": "string",
+          "description": "The URI of the file containing the user (etc/passwd) information. This field must be populated when 'usernameSource' is set to 'File'."
+        },
+        "ldapServer": {
+          "type": "string",
+          "description": "The fully qualified domain name or IP address of the LDAP server to use."
```

Comment:
primary and secondary IP addresses are only for AD.
ldapServer can be by address or by name.
You can specify file or ldap username download without AD.
Only AD username download requires the activeDirectory structure filled.

Reply:
RE: ldapServer: if it can be specified by name, how can we specify which DNS to use to resolve the LDAP server's fully qualified domain name?

RE: file or LDAP without AD: We probably should have examples for these, too, I think.

Reply:
Reactivating. Need to address the two items in my last comment.

Reply:
Updating examples based on the earlier discussion.

Reply:
Resolution is to not include DNS fields here as that may require an AOS change, which is not in scope. Resolving.